ibm_cloud_sdk_core/base_service.py

```diff
@@ -290,7 +290,8 @@ def prepare_request(self,
 
         if data and isinstance(data, dict):
             data = remove_null_values(data)
-            headers.update({'content-type': 'application/json'})
+            if headers.get('content-type') is None:
```

Comment:
There is a subtle problem here -- header names are actually _case insensitive_.  So you can't just check for `content-type` -- the user might have passed `Content-type`.

Reply:
Wow I didn't know about that. I'll make this case insensitive. Thanks!

Reply:
It looks like headers is a CaseInsensitiveDict instead of a regular dict, so it already makes all keys lowercase including the .get() function. So if the user passes in `Content-type` it still works here.